### PR TITLE
Fix upload when using organization as owner

### LINKED ISSUE #681: Attempt to create a new plugin under an org results in inconsistent state
I attempted to create a new plugin under the Nucleus namespace, Heisenberg. During the project creation, I set that the owner of the plugin should be the Nucleus org, which I own. I expected that the plugin would be created, and that I would be able to alter the documentation and upload new version as required.

However, this has not happened, instead, I've ended up with a plugin in a very weird state:

* The plugin can be seen under the org page at https://ore.spongepowered.org/Nucleus
* The plugin exists in the `dualspiral` namespace (see https://ore.spongepowered.org/dualspiral/Nucleus-Heisenberg)
* I have no editing or upload rights, as I am not listed as an owner of the plugin

I would expect that the plugin would be under the `Nucleus` namespace and that as the owner of the Nucleus org, I would inherit the permissions from the org with regards to this project unless otherwise specified.

This is using the latest version of Ore as of the time of writing, 7th Oct 2018 (which might be 1.6.14, but I'm not sure if there has been a deploy of this version)

### PR DESCRIPTION
**Pull request by felixoi**
This is not the best fix but it works and because no one has time to fix this properly we should go with this for now. Since upload is a key feature of Ore we should get this up afap.

Fixes #681

### CODE REVIEW

File: app/controllers/project/Projects.scala [hunk 1]
```diff
@@ -1,16 +1,11 @@
...
-
-import scala.collection.JavaConverters._
-import scala.concurrent.{ExecutionContext, Future}
-
-import play.api.cache.AsyncCacheApi
-import play.api.i18n.MessagesApi
-import play.api.libs.Files.TemporaryFile
-import play.api.mvc.{Action, AnyContent, MultipartFormData, Result}
 
+import _root_.util.StringUtils._
```
Comment (phase) [inline app/controllers/project/Projects.scala]: @Katrix How do you want to deal with import order?

File: app/models/project/ProjectSettings.scala [hunk 2]
```diff
@@ -70,13 +70,22 @@ case class ProjectSettings(
...
+      u.name
+    }
+
+    val updateProject = service.doAction(queryOwnerName.result).flatMap { ownerName =>
+      updateIfDefined(
+        project.copy(
+          category = Category.values.find(_.title == formData.categoryName).get,
+          description = noneIfEmpty(formData.description),
+          ownerId = formData.ownerId.getOrElse(project.ownerId),
+          ownerName = ownerName.head
```
Comment (phase) [inline app/models/project/ProjectSettings.scala:83]: Was this the main bug? We really need to gut names out soon.

File: app/models/user/User.scala [hunk 3]
```diff
@@ -141,22 +130,25 @@ case class User(
...
-                  m  <- memberTable if m.organizationId === o.id
-                  r  <- roleTable if m.userId === r.userId && r.organizationId === o.id
+                  p <- projectsTable if pScope.projectId.bind === p.id
+                  o <- orgaTable if p.userId === o.id
+                  m <- memberTable if m.organizationId === o.id && m.userId === this.userId.bind
+                  r <- roleTable if this.userId.bind === r.userId && r.organizationId === o.id
                 } yield r.roleType
 
-                service.doAction(query.to[Set].result).map(biggestRoleTpe)
+                service.doAction(query.to[Set].result).map { roleType =>
```
Comment (phase) [inline app/models/user/User.scala]: This variable should be called `roleTypes` since it's a list.

File: app/models/user/User.scala [hunk 4]
```diff
@@ -141,22 +130,25 @@ case class User(
...
+                  if (roleType.contains(RoleType.OrganizationAdmin) || roleType.contains(RoleType.OrganizationOwner)) {
```
Comment (phase) [inline app/models/user/User.scala]: Why are we not always calling `biggestRoleType` here? Shouldn't that method be the one that checks for Org roles?

**Review by phase — COMMENTED** [review]
Ref #681

File: app/models/project/ProjectSettings.scala (hunk 2 above)
Comment (felixoi) [reply app/models/project/ProjectSettings.scala:83]: This was the bug which created the project under the wrong namespace, yes.

File: app/controllers/project/Projects.scala (hunk 1 above)
Comment (Katrix) [reply app/controllers/project/Projects.scala]: It's all in the contributing document.  https://github.com/SpongePowered/Ore/blob/master/CONTRIBUTING.md#import-formatting

File: app/models/user/User.scala (hunk 4 above)
Comment (felixoi) [reply app/models/user/User.scala]: No. We only want to inherit permissions from orgs if it's an orga owner or an orga admin. `biggestRoleTpe` is a method on the `User` class which gets the role with the highest trust, it's more general and not only for use in this case.

**Review by phase — APPROVED** [review]

**Pull request merged**